src/Worker/ReschedulingDecorator.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neighborhoods\KojoWorkerDecoratorComponent\Worker;
+
+use DateTime;
+use Doctrine\DBAL\Connection;
+use LogicException;
+use Neighborhoods\KojoWorkerDecoratorComponent\WorkerInterface;
+use Neighborhoods\ThrowableDiagnosticComponent\ThrowableDiagnostic;
+use OutOfBoundsException;
+use Throwable;
+
+final class ReschedulingDecorator implements ReschedulingDecoratorInterface
+{
+    use DecoratorTrait;
+    use ThrowableDiagnostic\Builder\Factory\AwareTrait;
+
+    public const MAX_RESCHEDULE_DELAY_SECONDS = 86400;// (24 hours)
+
+    private /*int*/ $rescheduleDelaySeconds;
+    private /*string*/ $jobTypeCode;
+    private /*Connection*/ $connection;
+
+    public function work(): WorkerInterface
+    {
+        $this->runWorker();
+
+        if ($this->getApiV1WorkerService()->isRequestApplied()) {
+            $this->getApiV1WorkerService()
+                ->getLogger()
+                ->warning('Kojo worker state request already applied, not rescheduling.');
+        } else {
+            $this->reschedule();
+        }
+
+        return $this;
+    }
+
+    private function reschedule(): ReschedulingDecoratorInterface
+    {
+        $this->getConnection()->beginTransaction();
+        try {
+            $this->getApiV1WorkerService()
+                ->getNewJobScheduler()
+                ->setJobTypeCode($this->getJobTypeCode())
+                ->setWorkAtDateTime((new DateTime())->add(
+                    new \DateInterval('PT' . $this->getRescheduleDelaySeconds() . 'S')
+                ))
+                ->save();
+
+            $this->getApiV1WorkerService()
+                ->requestCompleteSuccess()
+                ->applyRequest();
+
+            $this->getConnection()->commit();
+        } catch (Throwable $throwable) {
+            $this->getConnection()->rollBack();
+            $this->getThrowableDiagnosticBuilderFactory()
+                ->create()
+                ->build()
+                ->diagnose($throwable);
+        }
+
+        return $this;
```

Comment:
**blocker**

As it stands, a `\Throwable` will mean that no job state change will be applied by this worker, userspace will exit, and kojo will complain, not knowing whether to `complete_success`, `complete_failed`, `panic`, etc. this job

Reply:
The `diagnose` method is expected to throw an exception.
This is only one of the decorators wrapping the worker. It's expected that the [ExceptionHandling](https://github.com/neighborhoods/KojoWorkerDecoratorComponent/blob/master/src/Worker/ExceptionHandlingDecorator.php) decorator wraps the others.
We don't want to mix exception handling into this, but let exceptions bubble up. What ReschedulingDecorator does is investigate if the exception is transient or not. When doing so only postgres related diagnostics is applied as indicated [here](https://github.com/neighborhoods/KojoWorkerDecoratorComponent/blob/rescheduling/src/Worker/ReschedulingDecorator/ThrowableDiagnostic/Builder.service.yml#L9).